projects/Mallard/src/hooks/use-net-info.tsx

```diff
@@ -93,8 +93,11 @@ export class NetInfoStateContainer {
     }
 
     subscribe(fn: (state: NetInfoState) => void) {
-        this.subscribers.push(fn)
-        fn(this.state)
+        this.fetch()
+            .then(fn)
+            .finally(() => {
+                this.subscribers.push(fn)
```

Comment:
the push itself should be done synchronously no?

Reply:
Also there was no call to `fn` originally in that function, why do we now need to call `fn`?